### PR TITLE
HIP-584: Copy ApproveAllowancePrecompile

### LINKED ISSUE #5652: HIP-584: Copy implementation for approve precompiles
### Problem

We need default implementations for precompiles related to approve logic, so we should copy needed logic from hedera-services into hedera-mirror-web3.

### Solution

Copy precompile logic for the following precompiles:

- ApprovePrecompile

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:  
We need default implementations for precompiles related to approve logic, so we should copy needed logic from hedera-services into hedera-mirror-web3.

**Related issue(s)**:

Fixes #5652

**Notes for reviewer**:  
Deleted `Store` from `Precompile` interface as it can be accessed from the `MessageFrame`

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java [hunk 1]
```diff
@@ -185,6 +185,10 @@ public Store getStore() {
         return store;
     }
 
+    public AccountAccessor getAccountAccessor() {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java:188]: I think it's not a good idea to expose the AccountAccessor outside of the StoreImpl. The Store should serve as a facade and hide all of the details.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/ApproveResult.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.store.contracts.precompile.codec;
+
+import com.hederahashgraph.api.proto.java.TokenID;
+
+public record ApproveResult(TokenID tokenId, boolean isFunguble) implements RunResult {}
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/ApproveResult.java:21]: isFunguble -> isFungible

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java [hunk 3]
```diff
@@ -208,35 +213,66 @@ protected Bytes computeInternal(final MessageFrame frame) {
...
+
+                this.precompile =
+                        precompileMapper.lookup(nestedFunctionSelector).orElseThrow();
+
+                if (AbiConstants.ABI_ID_HRC_ASSOCIATE == nestedFunctionSelector
+                        || AbiConstants.ABI_ID_HRC_DISSOCIATE == nestedFunctionSelector) {
+                    this.transactionBody = precompile.body(input, aliasResolver, new HrcParams(tokenId, senderAddress));
+                }
+                break;
+            case AbiConstants.ABI_ID_APPROVE, AbiConstants.ABI_ID_APPROVE_NFT, AbiConstants.ABI_ID_ERC_APPROVE:
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java:242]: We should treat differently the 3 precompiles.

ABI_ID_APPROVE and ABI_ID_APPROVE_NFT shouldn't need to set tokenId, where as ABI_ID_ERC_APPROVE should set it in the ApproveParams.

You can bring the specific logic for ABI_ID_ERC_APPROVE inside the redirect case. The other two signatures don't requite all of the logic related to reading redirect target and nestedFunctionSelector. Their logic could be placed in a different switch.

File: hedera-mirror-rest/viewmodel/index.js [hunk 4]
```diff
@@ -69,5 +69,5 @@ export {
   TransactionIdViewModel,
   BlockViewModel,
   FeeScheduleViewModel,
-  ContractStateViewModel
+  ContractStateViewModel,
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-rest/viewmodel/index.js:72]: Remove `hedera-mirror-rest` and `hedera-mirror-grpc` changes

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/ApproveResult.java (hunk 2 above)
Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/ApproveResult.java]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java (hunk 3 above)
Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java]: Modified

**Comment by 0xivanov** [comment]
Closing this PR, moving changes to #6446 , as this PR is too big to review properly

**Pull request closed without merging**